StoreMessageContentActivity/handler.ts

```diff
@@ -53,13 +67,107 @@ export type StoreMessageContentActivityResult = t.TypeOf<
   typeof StoreMessageContentActivityResult
 >;
 
+export const ServicePreferenceError = t.interface({
+  kind: t.keyof({ ERROR: null, LEGACY: null }),
+  message: t.string
+});
+
+export type ServicePreferenceError = t.TypeOf<typeof ServicePreferenceError>;
+
+export type ServicePreferenceValueOrError = (params: {
+  readonly serviceId: NonEmptyString;
+  readonly fiscalCode: FiscalCode;
+  readonly userServicePreferencesMode: ServicesPreferencesMode;
+  readonly userServicePreferencesVersion: number;
+}) => TaskEither<
+  ServicePreferenceError,
+  ReadonlyArray<BlockedInboxOrChannelEnum>
+>;
+
+type ServicePreferencesValues = Omit<
+  ServicePreference,
+  "serviceId" | "fiscalCode" | "settingsVersion"
+>;
+
+const channelToBlockedInboxOrChannelEnum: {
+  readonly [key in keyof ServicePreferencesValues]: BlockedInboxOrChannelEnum;
+} = {
+  isEmailEnabled: BlockedInboxOrChannelEnum.EMAIL,
+  isInboxEnabled: BlockedInboxOrChannelEnum.INBOX,
+  isWebhookEnabled: BlockedInboxOrChannelEnum.WEBHOOK
+};
+
+const servicePreferenceToBlockedInboxOrChannels: (
+  servicePreference: ServicePreference
+) => ReadonlyArray<BlockedInboxOrChannelEnum> = servicePreference =>
+  Object.entries(servicePreference)
+    // take only attributes of ServicePreferencesValues
+    .filter(([name, _]) => channelToBlockedInboxOrChannelEnum[name])
+    // take values set to false
+    .filter(([_, isEnabled]) => !isEnabled)
+    // map to BlockedInboxOrChannelEnum
+    .map(([name, _]) => channelToBlockedInboxOrChannelEnum[name]);
+
+const getServicePreferenceValueOrError = (
+  servicePreferencesModel: ServicesPreferencesModel
+): ServicePreferenceValueOrError => ({
+  fiscalCode,
+  serviceId,
+  userServicePreferencesMode,
+  userServicePreferencesVersion
+}): TaskEither<
+  ServicePreferenceError,
+  ReadonlyArray<BlockedInboxOrChannelEnum>
+> =>
+  taskEither
+    .of<ServicePreferenceError, ServicesPreferencesMode>(
+      userServicePreferencesMode
+    )
+    .chain(
+      fromPredicate(
+        _ => _ !== ServicesPreferencesModeEnum.LEGACY,
+        () => ({
+          kind: "LEGACY",
+          message: "User service preferences mode is LEGACY"
+        })
+      )
+    )
+    .map(() =>
+      makeServicesPreferencesDocumentId(
+        fiscalCode,
+        serviceId,
+        userServicePreferencesVersion as NonNegativeInteger
+      )
+    )
+    .chain(documentId =>
+      servicePreferencesModel
+        .find([documentId, fiscalCode])
+        .mapLeft<ServicePreferenceError>(failure => ({
+          kind: "ERROR",
+          message: `COSMOSDB|ERROR=${failure.kind}`
+        }))
```

Comment:
Instead of wrapping `CosmosErrors` into `ServicePreferenceError` type, maybe it's more eloquent to have them in union:
```ts
// the following it's just an example, you got the idea
type InvalidModeError = {
  kind: "INVALID_MODE", // what you call "LEGACY"
  value: ServicesPreferencesModeEnum
}

type ServicePreferenceError = InvalidModeError | CosmosErrors
```

Benefits are:
* you do not have manually re-map `CosmosErrors`
* the signature is more explicit about what kind of failure can happen



I made something similar [here](https://github.com/pagopa/io-functions-services/blob/2df422b7ba2e582f0a4b8e4030a0b6a854b3c154/utils/profile.ts#L104)

Reply:
Seems legit.

Reply:
Present in https://github.com/pagopa/io-functions-services/pull/142/commits/b50a4f98a99bbfe99da4326619f7b1434660e716